src/Protocols.Abstractions/Features/IConnectionTransportFeature.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Protocols.Features
+{
+    public interface IConnectionTransportFeature
```

Comment:
I also preserved the semantics for now but I think we should let the transport create the pipe but we'd have to invert the flow and have the kestrel provide the schedulers. More thought is required here.